R/bundle.R

```diff
@@ -222,15 +222,28 @@ enforceBundleLimits <- function(appDir, totalSize, totalFiles) {
 bundleApp <- function(appName, appDir, appFiles, appPrimaryDoc, assetTypeName,
                       contentCategory, verbose = FALSE, python = NULL,
                       condaMode = FALSE, forceGenerate = FALSE, quarto = NULL,
-                      isShinyApps = FALSE) {
+                      isShinyApps = FALSE, metadata = list()) {
   logger <- verboseLogger(verbose)
 
+  quartoInfo <- inferQuartoInfo(
+    appDir = appDir,
+    appPrimaryDoc = appPrimaryDoc,
+    quarto = quarto,
+    metadata = NULL
+  )
+  if (is.null(contentCategory) && !is.null(quartoInfo)) {
+    # Connect doesn't distinguish between Quarto projects and single
+    # documents, so neither will we. Quarto content are always deployed in
+    # "site" mode when using quarto::quarto_publish_app.
+    contentCategory <- "site"
```

Comment:
Is this forcing a content category onto Shiny Quarto?

On reflection, this feels too heavy-handed; maybe avoid marking site when we do not have a `_quarto.yml`?

Reply:
I agree that this feels heavy-handed.

I had considered the option you suggest, but I decided against it for two reasons: (1) It would feel weird to have a document's treatment change based on the presence of `_quarto.yml` file, rather than the contents of the metadata in that file. (2) `quarto::quarto_publish_app()` marks everything as "site".

An alternative idea I considered was setting it based on the output type. I guess I'd include a flag in the output of `inferQuartoInfo()` if the content type is "website" or "book".

If you think is a better alternative, I'll do that, otherwise, I'll just set it based on the presence `.quarto.yml`.

Reply:
`quarto::quarto_publish_app` does not set the content category; that happens only when using `quarto::quarto_publish_site`.
https://github.com/quarto-dev/quarto-r/blob/e06d0967033346ffd700b2e43f249aa4deb49893/R/publish.R#L233-L242

`writeManifest` is, in some ways, similar to `deployApp` -- the content category is provided by the caller. This is the current state for R Markdown sites, I believe.

Reply:
Then it must be push-button publishing that sets `contentCategory`, and that's a whole other thing.

I'll just remove this behavior, yep.

Reply:
Thanks. I agree that it's awkward that a caller needs to supply this information, but `writeManifest`, similar to `deployApp` sometimes doesn't have all the context it needs.